### PR TITLE
Cleanup documentation

### PR DESCRIPTION
**Pull request by arcalinea**
Start cleaning up unused/obsolete documentation. ref #1500

### CODE REVIEW

**Comment by bitcartel** [comment]
@arcalinea Is this for review or still a work in progress?

File: doc/release-notes.md [hunk 1]
```diff
@@ -1,219 +0,0 @@
-*** Warning: This document has not been updated for Zcash and may be inaccurate. ***
```
Comment (daira) [inline doc/release-notes.md]: Move this to doc/release-notes/release-notes-0.11.2.md rather than deleting it.

Comment (daira) [inline doc/release-notes.md]: Also, rename the doc/release-notes directory to doc/bitcoin-release-notes.

**Comment by bitcartel** [comment]
@arcalinea See comments above.  Upstream releated documentation should be preserved/archived rather than be deleted.

**Comment by arcalinea** [comment]
@bitcartel this is a work in progress, wanted some feedback on what direction to go with it.

**Comment by daira** [comment]
ACK. @zkbot r+

**Pull request merged**